recipe/gen_patch_json.py

```diff
@@ -2750,6 +2750,16 @@ def _gen_new_index_per_key(repodata, subdir, index_key):
             ):
                 _pin_stricter(fn, record, "numpy", "x", upper_bound="1.24")
 
+        # cuda-cccl and cuda-cccl-impl (released with CUDA 12) ship the same
+        # files as existing thrust/cub packages. These should conflict.
+        # Eventually users of thrust/cub should migrate to cuda-cccl[-impl]
+        if record_name in {"thrust", "cub"}:
+            new_constrains = record.get('constrains', [])
+            new_constrains.append("cuda-cccl <0.0.0a0")
+            new_constrains.append("cuda-cccl-impl <0.0.0a0")
+            new_constrains.append(f"cuda-cccl_{subdir} <0.0.0a0")
```

Comment:
You're correct that the thrust/cub packages don't clobber files in `cuda-cccl_{subdir}` (which installs to `targets/.../include`). They only clobber `cuda-cccl-impl` and `cuda-cccl`. However, I left this in because @kkraus14 suggested it, and it might not be a good idea to allow thrust/cub to be installed together with CUDA 12 packages. The `thrust`/`cub` versions shipped on conda-forge don't have official support for CUDA 12, and `cuda-cccl_{subdir}` is specific to CUDA 12. I would be comfortable saying this is an unsupported case that would fall in the same bucket as `cudatoolkit` (CUDA 11) not being installable at the same time as `cuda-nvcc` (CUDA 12). What do you think?